### DISCUSSION TITLE
Barnard 测试网挖矿指南| Mining Guide

### DISCUSSION THREAD

**Discussion by sanlee42** [starcoinorg/starcoin#2327] [Testnet]
[Starcoin 挖矿比赛第二期-Barnard 测试网暨主网上线预告](https://news.starcoin.org/zh/2021/starcoin_testnet_mining_competition_r2/)
[Mining Competition Phase 2 - Barnard Test Network and Mainline Launch Preview](https://news.starcoin.org/en/2021/starcoin_testnet_mining_competition_r2/)

### 下载 && 安装

在 release 页面根据自己的操作系统下载 [starcoin-v1.0.0-beta.6](https://github.com/starcoinorg/starcoin/releases/tag/v1.0.0-beta.6) 版本的二进制包，或者用 docker  镜像
```
docker pull starcoin/starcoin:v1.0.0-beta.6
```

如果想了解其他安装方式请参看[安装](https://developer.starcoin.org/zh/setup/install/) 文档。

### 启动节点加入 Starcoin barnard 测试网络

启动时需要配置一个 metrics 采集地址，用于 starcoin 节点性能监控，同时也是一种节点运行证明，会用在节点抽奖统计的数据中。
> 注：如果 metrics 服务出现错误 Msg("error sending request for url (http://miner-metrics-pushgw.starcoin.org:9191/metrics/job/starcoin_push/instance/vps-00abdb26): operation timed out") 请忽略，我们有其他方式统计节点存活，这个不是必须的。

```
starcoin -n barnard --push-server-url http://miner-metrics-pushgw.starcoin.org:9191/ --push-interval 60 --miner-thread 1
```

通过 docker 启动：
```
docker run --name starcoin-barnard -d --network host -v /data/starcoin/:/data/starcoin/ starcoin/starcoin:v1.0.0-beta.6 /starcoin/starcoin -n barnard -d /data/starcoin --push-server-url http://miner-metrics-pushgw.starcoin.org:9191/ --push-interval 60 --miner-thread 1
```

> 注: 
> 1. --miner-thread 参数可用来指定挖矿线程，请根据自己节点的配置信息调整。
> 2. 数据目录默认在 $home/.starcoin ，可以通过 -d 参数制定。docker 的例子中数据目录放置在 /data/starcoin 下。
> 3. Windows 用户请讲 starcoin 替换成 starcoin.exe
> 4. 如果有多台节点在同一个局域网中，可以加上 --discover-local true 参数，这样同一个局域网的节点会在第一时间互相发现。

更多信息请参看文档：[启动和加入网络](https://developer.starcoin.org/zh/setup/runnetwork/)。

### 挖矿客户端

节点启动后，可以通过接入额外的挖矿客户端来连接到节点挖矿，请参看 https://developer.starcoin.org/zh/cli/mint/ 。

### 使用命令行查看节点信息

使用以下命令进入控制台：

```
starcoin --connect ~/.starcoin/barnard/starcoin.ipc console 
```

在 starcoin console 下输入:

   ``` 
   starcoin% help # 查看所有命令
   starcoin% account show # 查看账号信息，如果 balance 不为空，说明挖出块了。
   starcoin% account export <account_address> -p <password> # 导出 private key
   starcoin% account change-password 修改密码
   starcoin% node sync progress # 查看同步进度
   ```

更多命令请参考[开发者手册](http://developer.starcoin.org) 中的 [命令行章节](https://developer.starcoin.org/zh/cli/) 。 

### 通过命令行工具发起交易

请参看文档 [第一笔链上交易](https://developer.starcoin.org/zh/cli/first_transaction/)

## 使用桌面客户端(建议新手)

1. 下载对应系统的版本(Support MacOs,Windows)：[Github 下载链接](https://github.com/starcoinorg/starcoin_node_gui/releases/) 
2. 解压后点击 starcoin_node 启动
3. 弹出安全提示窗口，点击：「好」
4. 打开电脑 系统偏好设置（齿轮图标）-->安全性与隐私--> 点击「仍要打开」（页面右下角）
5. 弹出安全提示窗口，点击：「打开」
6. Starcoin 挖矿程序窗口：创建昵称-->确认-->点击小三角即可开始挖矿。

![image](https://user-images.githubusercontent.com/77268/112781826-35edfb80-907e-11eb-9209-70a14e8bb120.png)

Windows版本启动程序后，跳出安全弹窗点击确认即可。

>注: 当前的算力已经比较高，单机挖矿已经很难挖出来了，可以期待矿池版本。

其他问题请回复本帖或者加入 Discord: https://discord.gg/DJkU8NAezd

### FAQ

1. 服务器最低配置是？ 2G memory, 2core cpu, 20G disk.
2. 挖矿的地址和账号如何创建？ 节点启动时会自动创建一个账号和地址，挖矿使用默认账号，密码默认为空。启动后请登录到 console 修改密码以及导出私钥进行备份。
3. STC 的单位是？STC 的小数点后有 9 位。
4. 账户余额为空是什么意思？账户余额为空说明该账号尚未在链上创建。账户余额为 0 说明账号已经创建，但奖励尚未到账，区块奖励要延迟 7 个区块到账。
4. 为什么日志显示挖出了块，但账户余额为空？因为日志显示挖出来块只是说明挖矿算法要求的谜题给解出来了，节点需要把区块广播到网络上，竞争成为最长链，才能得到区块奖励。如果广播的时候其他节点先广播了，或者该区块所在的链没有竞争成为最长链，则不能得到奖励。
5. 为什么我的账号本来显示余额，但后来变为了空？原因同上，链重组(reorg)了，挖出的区块没有在最长链，所以奖励也没了。
6. 出现错误 Try to connect node by ipc: "$home/.starcoin/barnard/starcoin.ipc" Starcoin cmd exits abnormally: Connection refused (os error 61) 是什么意思？节点未能正常退出，请把 $home/.starcoin/barnard/starcoin.ipc 文件给删掉。

## Starcoin 矿机挖矿
参考: https://starcoin-miner.gitbook.io/starcoin-miner-guide/

**Comment by jiangydev** [comment]
运行 Starcoin 节点系列教程（有图）：
含 Windows，Docker：
https://www.yuque.com/books/share/e0f48d8f-62d1-4602-91d1-08a150958efe
已迁移：https://starcoin-miner.gitbook.io/starcoin-miner-guide/

**Comment by talonliu** [comment]
更新一下，Windows10命令行图文版手册
[StarCoin Barnard 测试网简易挖矿指南-2021-4-1](https://note.youdao.com/ynoteshare1/index.html?id=a4cbe1d71c30f64ff08e070139355fe8&type=note)

**Comment by talonliu** [comment]
[StarCoin区块数据批量复制](http://note.youdao.com/noteshare?id=a8507e9bd511a08a3b1f3f856bf039f1&sub=DDF7BA39CC4142F49648F584B3C617C9)

**Comment by thisjim** [comment]
获得的矿机有使用指南吗？

**Reply by sanlee42** [reply]
https://starcoin-miner.gitbook.io/starcoin-miner-guide/

**Comment by caodongjian** [comment]
为啥 我余额一直为空？有什么办法处理

**Reply by jolestar** [reply]
先确认区块是否同步完成。如果同步完成那说明就是确实没挖出来。
